ShelterModule/Controllers/PetController.cs

```diff
@@ -75,7 +78,12 @@ public async Task<ActionResult<PetResponse>> Post(PetCreationRequest request)
         if (await _validator.ValidateClaims(User) is not TokenValidationResult.Valid)
             return Unauthorized();
 
-        var pet = Pet.FromRequest(request, User.GetId());
+        var shelter = await _shelterQuery.GetByIdAsync(User.GetId());
+
+        if (shelter == null)
+            return Unauthorized();
+
+        var pet = Pet.FromRequest(request, shelter);
```

Comment:
We could move this to `AddAsync()` and return `Result` when shelter is not found